src/guiguts/tools/jeebies.py

```diff
@@ -0,0 +1,297 @@
+"""Jeebies check functionality"""
+
+import importlib.resources
+import logging
+import regex as re
+
+from guiguts.data import dictionaries
+from guiguts.checkers import CheckerDialog
+from guiguts.maintext import maintext
+
+logger = logging.getLogger(__package__)
+
+DEFAULT_DICTIONARY_DIR = importlib.resources.files(dictionaries)
+PARANOID_LEVEL_3WORDS = 1.0
+
+_the_jeebies_checker = None  # pylint: disable=invalid-name
+
+
+class DictionaryNotFoundError(Exception):
+    """Raised when no hebe-forms dictionary found."""
+
+    def __init__(self, file: str) -> None:
+        self.file = file
+
+
+class JeebiesChecker:
+    """Provides jeebies check functionality."""
+
+    def __init__(self) -> None:
+        """Initialize SpellChecker class."""
+        self.dictionary: dict[str, int] = {}
+        self.load_phrases_file_into_dictionary()
+
+    def check_for_jeebies_in_file(self) -> None:
+        """Check for jeebies in the currently loaded file."""
+
+        # Create the checker dialog to show results
+        checker_dialog = CheckerDialog.show_dialog(
+            "Jeebies Results", rerun_command=jeebies_check
+        )
+        checker_dialog.reset()
+
+        # Get the whole of the file from the main text widget
+        text = maintext().get_text()
```

Comment:
Since you don't need `text` later, consider:
```python
input_lines = maintext().get_text().splitlines()
```

This makes it so we only have one copy of the text in memory -- the array form.